### PR TITLE
Update oathkeeper Helm chart to 0.23.2

### PR DESCRIPTION
**Pull request by barchw**
**Description**

Changes proposed in this pull request:

- Sync Oathkeeper chart with latest ory/k8s release
- Move security context for sidecar and controller to values.yaml
- Annotate Kyma specific additions with a comment

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/14266

### CODE REVIEW

File: resources/ory/charts/oathkeeper/values.yaml [hunk 1]
```diff
@@ -1,109 +1,112 @@
...
+    # -- If you do want to specify additional labels, uncomment the following lines, adjust them as necessary, and remove the curly braces after 'labels:'.
+    # e.g.  app: oathkeeper
     labels: {}
-    #      If you do want to specify additional labels, uncomment the following
-    #      lines, adjust them as necessary, and remove the curly braces after 'labels:'.
-    #      e.g.  app: oathkeeper
 
-  # Configures the Kubernetes service for the api port.
+  # -- Configures the Kubernetes service for the api port.
   api:
```
Comment (cnvergence) [inline resources/ory/charts/oathkeeper/values.yaml:54]: I think we miss `service.api.name` value here

**Review by cnvergence — CHANGES_REQUESTED** [review]

File: resources/ory/charts/oathkeeper/values.yaml (hunk 1 above)
Comment (barchw) [reply resources/ory/charts/oathkeeper/values.yaml:54]: service.api.name = "" added with a comment

File: resources/ory/charts/oathkeeper/templates/NOTES.txt [hunk 2]
```diff
@@ -1,64 +0,0 @@
-{{ include "oathkeeper.check.override.consistency" . }}
```
[the remark this answers is not in the document]
Comment (barchw) [reply resources/ory/charts/oathkeeper/templates/NOTES.txt:1]: Demo files re-added for the purpose of compatibilty with official ory charts

File: resources/ory/charts/oathkeeper/values.yaml [hunk 3]
```diff
@@ -1,109 +1,114 @@
...
-    # En-/disable the service
+    # -- Name for the api port
+    name: ""
```
Comment (cnvergence) [inline resources/ory/charts/oathkeeper/values.yaml:56]: It should be named http

**Review by cnvergence — CHANGES_REQUESTED** [review]

**Review by cnvergence — APPROVED** [review]

**Pull request merged**